tests/unit/package_managers/bundler/test_parser.py

```diff
@@ -0,0 +1,189 @@
+    lockfile_path.path.touch()
+
+    return gemfile_path, lockfile_path
+
+
+MOCKED_BASE_JSON = textwrap.dedent(
+    """
+    {
+      "bundler_version": "2.5.10",
+      "dependencies": [
+        {
+          "name": "example",
+          "version": "0.1.0"
+        }
+      ]
+    }
+    """
+)
+
+
+def test_parse_lockfile_without_bundler_files(rooted_tmp_path: RootedPath) -> None:
+    with pytest.raises(PackageRejected) as exc_info:
+        parse_lockfile(rooted_tmp_path)
+
+    assert (
+        "Gemfile and Gemfile.lock must be present in the package directory"
+        in exc_info.value.friendly_msg()
+    )
+
+
+@mock.patch("cachi2.core.package_managers.bundler.parser.run_cmd")
+def test_parse_lockfile_os_error(
+    mock_run_cmd: mock.MagicMock,
+    empty_bundler_files: tuple[RootedPath, RootedPath],
+    rooted_tmp_path: RootedPath,
+) -> None:
+    mock_run_cmd.side_effect = subprocess.CalledProcessError(returncode=1, cmd="cmd")
+
+    with pytest.raises(PackageManagerError) as exc_info:
+        parse_lockfile(rooted_tmp_path)
+
+    assert f"Failed to parse {empty_bundler_files[1]}" in exc_info.value.friendly_msg()
+
+
+@mock.patch("cachi2.core.package_managers.bundler.parser.run_cmd")
+@pytest.mark.parametrize(
+    "error, expected_error_msg",
+    [
+        ("LOCKFILE_INVALID_URL", "Input should be a valid URL"),
+        ("LOCKFILE_INVALID_URL_SCHEME", "URL scheme should be 'https'"),
+        ("LOCKFILE_INVALID_REVISION", "String should match pattern '^[a-fA-F0-9]{40}$'"),
+        ("LOCKFILE_INVALID_PATH", "PATH dependencies should be within the package root"),
+    ],
+)
+def test_parse_lockfile_invalid_format(
+    mock_run_cmd: mock.MagicMock,
+    error: str,
+    expected_error_msg: str,
+    empty_bundler_files: tuple[RootedPath, RootedPath],
+    rooted_tmp_path: RootedPath,
+) -> None:
+    mocked_json = json.loads(MOCKED_BASE_JSON)
+
+    if error == "LOCKFILE_INVALID_URL":
+        mocked_json["dependencies"][0].update(
+            {
+                "type": "git",
+                "url": "github",
+                "revision": "f" * 40,
+            }
+        )
+    elif error == "LOCKFILE_INVALID_URL_SCHEME":
+        mocked_json["dependencies"][0].update(
+            {
+                "type": "git",
+                "url": "http://github.com/3scale/json-schema.git",
+                "revision": "f" * 40,
+            }
+        )
+    elif error == "LOCKFILE_INVALID_REVISION":
+        mocked_json["dependencies"][0].update(
+            {
+                "type": "git",
+                "url": "https://github.com/3scale/json-schema.git",
+                "revision": "abcd",
+            }
+        )
+    elif error == "LOCKFILE_INVALID_PATH":
+        mocked_json["dependencies"][0].update(
+            {
+                "type": "path",
+                "path": "/root/pathgem",
+            }
+        )
+
+    mock_run_cmd.return_value = json.dumps(mocked_json)
```

Comment:
I think this is just a marginal improvement over the last revision (e.g. doing json.load only to do json.dumps later instead of using a dictionary in the first place, setting fields that are not being tested in a given parametrize run, etc.), but since this is supposed to be just a temporary solution anyway, I'll let it fly and not block this further.

Reply:
fixed